### ISSUE TITLE
get wrong srv records from kube-dns

### ISSUE THREAD

**Issue by zhashuyu** [kubernetes/kubernetes#51326]
**What happened**:
For some reason I need get srv records in kube-dns , below is what I got
```
[root@devo-ui-37mmn /]# nslookup
> set q=srv
> devo-ui.dev.svc.cluster.local.
Server:     10.254.0.2
Address:    10.254.0.2#53

devo-ui.dev.svc.cluster.local   service = 10 100 0 3339386563373339.devo-ui.dev.svc.cluster.local.
> 
```
host **devo-ui-37mmn** is one of my pod, the query string **devo-ui.dev.svc.cluster.local** is the service in kubernetes and **devo-ui-37mmn** is connect to the service.

I need get the srv records in **devo-ui.dev.svc.cluster.local** , but I got **3339386563373339.devo-ui.dev.svc.cluster.local**. Is this digital **3339386563373339** represent **devo-ui-37mmn**, and how do I get the normal srv records?

**How to reproduce it**:
Create some pods and service , then do nslookup, it will show this.

**Environment**:
- Kubernetes: 1.6.8 / 1.7.4
- kube-dns: k8s-dns-kube-dns-amd64:1.14.2
                   k8s-dns-dnsmasq-nanny-amd64:1.14.2
                   k8s-dns-sidecar-amd64:1.14.2

**this is what I got with windows ad**
```
$ nslookup
> set q=srv
> _ldap._tcp.example.com
Server:		10.110.200.19
Address:	10.110.200.19#53

_ldap._tcp.skyinno.com	service = 0 100 389 ad2.example.com.
_ldap._tcp.skyinno.com	service = 0 100 389 ad1.example.com.
> 
```

**Comment by cmluciano** [comment]
@kubernetes/sig-network-misc 

/area dns

**Comment by MrHohn** [comment]
@zhashuyu I think you will need to set `spec.hostname` on your pod in order to have the non-hashed hostname. Something like:
```
apiVersion: v1
kind: Pod
metadata:
  name: busybox1
spec:
  hostname: busybox1
```

**Comment by zhashuyu** [comment]
Sorry , I still got the hashed result
```
[root@master24 yaml_files]# cat busybox-pods.yaml
apiVersion: v1
kind: Pod
metadata:
  name: busybox1
  namespace: kube-system
  labels:
    app: busybox1
spec:
  #labels:
  #  app: busybox1
  hostname: busybox1
  containers:
  - image: hub.test.com/common/busybox:latest
    command:
      - sleep
      - "3600"
    imagePullPolicy: IfNotPresent
    resources:
      limits:
        cpu: 1
        memory: 10Mi
      requests:
        cpu: 0.1
    name: busybox1
  restartPolicy: Always
[root@master24 yaml_files]# cat busybox-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: busybox1
  namespace: kube-system
spec:
  ports:
  - port: 22
    name: busybox1
    targetPort: 22
    protocol: TCP
  selector:
    app: busybox1
[root@master24 yaml_files]# 
[root@master24 yaml_files]# kubectl-kube-system describe pods busybox1
Name:		busybox1
Namespace:	kube-system
Node:		10.110.200.21/10.110.200.21
Start Time:	Mon, 28 Aug 2017 09:12:39 +0800
Labels:		app=busybox1
Annotations:	<none>
Status:		Running
IP:		172.30.41.16
Controllers:	<none>
......
[root@master24 yaml_files]# kubectl-kube-system describe svc busybox1
Name:			busybox1
Namespace:		kube-system
Labels:			<none>
Annotations:		<none>
Selector:		app=busybox1
Type:			ClusterIP
IP:			10.254.93.185
Port:			busybox1	22/TCP
Endpoints:		172.30.41.16:22
Session Affinity:	None
Events:			<none>
[root@master24 yaml_files]# kubectl-kube-system exec -it nslookuphost /bin/bash
[root@nslookuphost /]# nslookup
> set q=srv
> busybox1.kube-system
Server:		10.254.0.2
Address:	10.254.0.2#53

busybox1.kube-system.svc.cluster.local	service = 10 100 0 6531656264646235.busybox1.kube-system.svc.cluster.local.
> quit
Server:		10.254.0.2
Address:	10.254.0.2#53
```

**Comment by bowei** [comment]
3339386563373339.devo-ui.dev.svc.cluster.local. is the A record for the pod, which you can resolve to the IP of the pod. This matches the behavior of standard SRV records as far as I know. Can you be more precise regarding what you were expecting to get?

**Comment by zhashuyu** [comment]
I just want get the IP or hostname of the pod from srv records, but I got **3339386563373339** instead, and I don't know how to convert **3339386563373339**  to IP or hostname.

**Comment by bowei** [comment]
`3339386563373339.devo-ui.dev.svc.cluster.local.` is an A record, so you need to resolve it to get the IP address. This is the same as your active directory example, as `ad2.example.com.` will also need to be resolved to get the IP address.

**Comment by chaudum** [comment]
We faced the same issue when we ran a StatefulSet with two Services, one regular (named `app-name-service`) and one headless (named `app-name-service-headless`).
Both services referenced the same Pods from the Statefulset with a selector `name=app-name` where `app-name` is the name of the StatefulSet.

However, the Statefulset's `spec.serviceName` referenced the `app-name`, not the `app-name-headless` Service.
After changing to the correct Service in the StatefulSet, the SRV record returned the lsit of expected hostnames, such as `app-name-0.namespace.svc.cluster.local`.
